add support for the kintex/virtex high performance banks

hansfbaier opened the pull request:
Signed-off-by: Hans Baier <hansfbaier@gmail.com>

hansfbaier commented:
@mkurc-ant @kgugala @mithro I cannot reproduce the failure for kintex7 locally.
Would you please send me the file ```/root/prjxray/prjxray/fuzzers/058-pip-hclk/logs_xc7k70tfbg676-2/stdout.2022-11-09T06-17-04.992001.log``` so that I can figure out what went wrong there.

hansfbaier commented:
@kgugala @mkurc-ant Any ideas why the kintex job was aborted?

pgielda commented:
I am guessing 6h timeout?

hansfbaier commented:
@pgielda Unfortunately it seems that kintex7 really needs that much time. I tried reducing the ROI, but then 025-bram-config fails complaining it has not enough BRAM instances in the ROI. What should I do now?

pgielda commented:
FYI @tmichalak

tmichalak commented:
@hansfbaier the default timeout for a job, before GitHub cancels it, is 360 minutes, which is exactly what seems to happen here.
You can change that using [timeout-minutes](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepstimeout-minutes) per job or step - in your case using that in the [run tests](https://github.com/kintex-chatter/prjxray/blob/k7-iob18/.github/workflows/Pipeline.yml#L76) step should solve the problem.

kgugala commented:
hmm, looks like the timeout setting was ignored?

tmichalak commented:
Looks to me that the `timeout-minutes` was added to the 'Build' step while the step that times out is the [run test](https://github.com/kintex-chatter/prjxray/blob/k7-iob18/.github/workflows/Pipeline.yml#L77) step. @hansfbaier

hansfbaier commented:
@kgugala @tmichalak Yes I added it to the wrong step. Thanks!

hansfbaier commented:
@tmichalak Now I have it on 'Run Tests' but unfortunately it did not work.

hansfbaier commented:
Looks like we have reached a hard limit on GitHub here
![image](https://user-images.githubusercontent.com/148607/201908654-dffae55d-a7f5-48c5-a0fd-98a18cec2a6f.png)
To have a job running more than 6 hours, we would need to have a self hosted agent, see
https://stackoverflow.com/questions/68187987/github-actions-job-timeout

kgugala commented:
@hansfbaier the CI here is run in custom runners

kgugala commented:
timeout seems to be working. Kintex job is now running for 7h

hansfbaier commented:
@kgugala Great! Moving the timeout configuration into the parent element did the trick!

hansfbaier commented:
@kgugala It looks like some of the other series tests are flickering. Race condition?

kgugala commented:
there might be some instabilities - in general fuzzers, by nature, random so sometimes we hit a case where the flow fails unexpectedly

hansfbaier commented:
Ah, the 037-iob18-pips fuzzer does not terminate, the todo's do not vanish:
```
RIOI.IOI_ODELAY0_CLKIN.IOI_OCLK_0
RIOI.IOI_ODELAY1_CLKIN.IOI_OCLK_1
RIOI.RIOI_O0.RIOI_ODELAY0_DATAOUT
RIOI.RIOI_O0.RIOI_OLOGIC0_OQ
RIOI.RIOI_O1.RIOI_ODELAY1_DATAOUT
RIOI.RIOI_O1.RIOI_OLOGIC1_OQ
```
These are the unsolved TODOs so far

hansfbaier commented:
Great! All the fuzzers ran successfully now, only checkdb still reports collisions. Got to figure those out.

hansfbaier commented:
My newly commited conditional in a Makefile might fail for non kintex. Will look into it tomorrow. Will be a trivial fix. Kintex is redy for review nevertheless.

hansfbaier commented:
Another Makefile needs the same adjustment. Also, fuzzers/036-iob18-ologic/bits.dbf I have just copy and pasted. Should this be empty by default? Is the fileformat documented somewhere?

tmichalak commented:
@hansfbaier the `.dbf` is for zero groups filtering and it can be empty by default. You can find a short documentation for this format [here](https://github.com/f4pga/prjxray/blob/master/utils/dbfixup.py#L83)

tmichalak commented on fuzzers/030-iob18/README.md:
```diff
@@ -0,0 +1,2 @@
+# IOB18 Fuzzer
```
Would be good to have something here

hansfbaier replied:
OK rebased to current master and review notes applied.

tmichalak reviewed (commented):
Please update the date in the license headers in the new files you're adding e.g. 2017-2022

tmichalak commented:
@hansfbaier CI is indeed green. Just had a two small comments. Other than that LGTM.

hansfbaier commented:
@tmichalak Thank you, will tidy up tomorrow!

hansfbaier commented:
@tmichalak I applied your review notes and rebased to master. Concerning the .dbf files I removed/adjusted the parts which are obviously not applicaple, but kept the rest of the bits, because in the fuzzer results it looks like the same frame range is used and thus I suspect by default that the changes are valid, especially as some of the io logic fuzzers gave identical results to the high range bank fuzzers so that I actually deleted them and just copied their bits over.

hansfbaier commented:
@tmichalak It is green again. Except Artix, but that seems to be that flickering timeout int the fuzzer
```
2022-11-23T22:21:04.8795880Z 22:21:04 | 2022-11-23T22:21:04 - xc7a200tffg1156-1/065b-gtp-common-pips      - 1h56m: + '[' -f /root/prjxray/prjxray/fuzzers/065b-gtp-common-pips/build_xc7a200tffg1156-1/todo/timeout ']'
2022-11-23T22:21:04.8796540Z 22:21:04 | 2022-11-23T22:21:04 - xc7a200tffg1156-1/065b-gtp-common-pips      - 1h56m: + echo 'ERROR: timeout'
2022-11-23T22:21:04.8797097Z 22:21:04 | 2022-11-23T22:21:04 - xc7a200tffg1156-1/065b-gtp-common-pips      - 1h56m: + exit 1
2022-11-23T22:21:04.8797705Z 22:21:04 | 2022-11-23T22:21:04 - xc7a200tffg1156-1/065b-gtp-common-pips      - 1h56m: make[2]: *** [../pip_loop.mk:68: run] Error 1
2022-11-23T22:21:04.8798403Z 22:21:04 | 2022-11-23T22:21:04 - xc7a200tffg1156-1/065b-gtp-common-pips      - 1h56m: 
2022-11-23T22:21:04.8799154Z 22:21:04 | 2022-11-23T22:21:04 - xc7a200tffg1156-1/065b-gtp-common-pips      - 1h56m: --------------------------------------------------------------------------
2022-11-23T22:21:04.8799847Z 22:21:04 | 2022-11-23T22:21:04 - xc7a200tffg1156-1/065b-gtp-common-pips      - 1h56m: !Failed! @ 2022-11-23T22:21:04.581598 with exit code: 2
```

tmichalak commented:
@hansfbaier nice work. Thanks a lot !
